test/leafnode_test.go

```diff
@@ -3385,3 +3389,165 @@ func TestServiceExportWithMultipleAccounts(t *testing.T) {
+
+		accounts: {
+		    INTERNAL: {
+		        users: [
+		            {user: good, password: pwd}
+		        ]
+		        exports: [{service: "foo", response: singleton}]
+			    imports: [
+		            {
+		                service: {
+		                    account: EXTERNAL
+		                    subject: "evilfoo"
+		                }, to: from_evilfoo
+		            }
+		        ]
+		    },
+		    EXTERNAL: {
+		        users: [
+		            {user: evil, password: pwd}
+		        ]
+		        exports: [{service: "evilfoo", response: singleton}]
+			    imports: [
+		            {
+		                service: {
+		                    account: INTERNAL
+		                    subject: "foo"
+		                }, to: goodfoo
+		            }
+		        ]
+		    }
+		}
+	`))
+	defer os.Remove(confG)
+
+	srvG, optsG := RunServerWithConfig(confG)
+	defer srvG.Shutdown()
+
+	eConfigTemplate := `
+		server_name: E
+		listen: 127.0.0.1:-1
+		leafnodes {
+			listen: "127.0.0.1:-1"
+			remotes = [
+			{
+				url:"nats://127.0.0.1:%d"
+				account:"EXTERNAL_GOOD"
+			}
+			]
+		}
+
+		accounts: {
+		    INTERNAL_EVILINC: {
+		        users: [
+		            {user: evil, password: pwd}
+		        ]
+		        exports: [{service: "foo", response: singleton}]
+			    imports: [
+		            {
+		                service: {
+		                    account: EXTERNAL_GOOD
+		                    subject: "goodfoo"
+		                }, to: from_goodfoo
+		            }
+		        ]
+		    },
+		    EXTERNAL_GOOD: {
+		        users: [
+		            {user: good, password: pwd}
+		        ]
+		        exports: [{service: "goodfoo", response: singleton}]
+			    imports: [
+		            {
+		                service: {
+		                    account: INTERNAL_EVILINC
+		                    subject: "foo"
+		                }, to: evilfoo
+		            }
+		        ]
+		    },
+		}
+	`
+
+	confE := createConfFile(t, []byte(fmt.Sprintf(eConfigTemplate, optsG.LeafNode.Port)))
+	defer os.Remove(confE)
+
+	srvE, optsE := RunServerWithConfig(confE)
+	defer srvE.Shutdown()
+
+	// connect to confE, and offer a service
+	nc2, err := nats.Connect(fmt.Sprintf("nats://evil:pwd@%s:%d", optsE.Host, optsE.Port))
+	if err != nil {
+		t.Fatalf("Error on connect: %v", err)
+	}
+	defer nc2.Close()
+
+	nc2.Subscribe("foo", func(msg *nats.Msg) {
+		if err := msg.Respond([]byte("world")); err != nil {
+			t.Fatalf("Error on respond: %v", err)
+		}
+	})
+	nc2.Flush()
+
+	nc, err := nats.Connect(fmt.Sprintf("nats://good:pwd@%s:%d", optsG.Host, optsG.Port))
+	if err != nil {
+		t.Fatalf("Error on connect: %v", err)
+	}
+	defer nc.Close()
+
+	resp, err := nc.Request("from_evilfoo", []byte("hello"), 2*time.Second)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if resp == nil || strings.Compare("world", string(resp.Data)) != 0 {
+		t.Fatal("Did not receive the correct message")
+	}
+
+	// Now restart server E and requestor and replier.
+	srvE.Shutdown()
+	nc.Close()
```

Comment:
Why do you close clients here? But if you have to do it, I would close them (especially nc2) before shutting down srvE so that nc2 does not attempt to reconnect.
If it is simply to speed up the test, you could pass a smaller reconnect wait when creating the connections.

Reply:
Was following the description @matthiashanel had put in the issue that both the replier and requestor were restarted. I could move then closes, won't affect test.

Reply:
No problem, I was just wondering.

Reply:
@derekcollison restart or not did not matter to the outcome... Not knowing what the issue was, I noted it.

Reply:
Fixed.